samtranslator/model/sam_resources.py

```diff
@@ -415,6 +412,51 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def] # noqa: P
 
         return resources
 
+    def _make_lambda_role(
+        self,
+        lambda_function: LambdaFunction,
+        intrinsics_resolver: IntrinsicsResolver,
+        execution_role: IAMRole,
+        resources: List[Any],
+        conditions: Dict[str, Any],
+    ) -> None:
+        lambda_role = lambda_function.Role
+
+        if lambda_role is None:
+            resources.append(execution_role)
+            lambda_function.Role = execution_role.get_runtime_attr("arn")
+
+        if is_intrinsic_if(lambda_role):
```

Comment:
I don't know how much we save in time, but I think for clarity, this could be an `elif ` to make it clear that it's a different case that the one right above.